app/vmalert/manager.go

```diff
@@ -100,9 +100,11 @@ func (m *manager) startGroup(ctx context.Context, g *Group, restore bool) error
 			select {
 			case <-ctx.Done():
 				sleepTimer.Stop()
+				close(g.finishedCh)
 				return
 			case <-g.doneCh:
 				sleepTimer.Stop()
+				close(g.finishedCh)
```

Comment:
Could it panic? Looks like channel could be closed twice:

```
start(ctx context.Context, nts func() []notifier.Notifier, rw *remotewrite.Client, rr datasource.QuerierBuilder) {
	defer func() { close(g.finishedCh) }()
```

Reply:
I agree it looks not rly cool, but there will be no panic.
This chan can be closed either here or inside of `group.start()` func which is called inside of this routine. So the execution here is sequential.
Alternative approach would be only maintaining this chan inside this one function, but then `group.start()` will be harder to use in tests.

Reply:
@f41gh7 please see PR updated